evm/test/AccessGrants.js

```diff
@@ -24,21 +24,39 @@ describe("AccessGrants", function () {
 
   describe("Grant management", function () {
     describe("Creating grants", function () {
-      it("A grant can only be owned by its creator", async function () {
-        const { accessGrants, signer1: owner, signer2: grantee } = await loadFixture(deployAndPopulateContractFixture);
+      describe("By owner through insertGrant", function () {
+        it("A grant can only be owned by its creator", async function () {
+          const { accessGrants, signer1: owner, signer2: grantee } = await loadFixture(deployAndPopulateContractFixture);
 
-        let grants = await accessGrants.findGrants(owner, grantee, "1A");
+          let grants = await accessGrants.findGrants(owner, grantee, "1A");
 
-        expect(grants.length).to.equal(1);
-        expect(grants[0].owner).to.equal(owner.address);
+          expect(grants.length).to.equal(1);
+          expect(grants[0].owner).to.equal(owner.address);
+        });
+
+        it("A duplicate grant cannot be created", async function () {
+          const { accessGrants, signer1: owner, signer2: grantee } = await loadFixture(deployAndPopulateContractFixture);
+
+          await expect(
+            accessGrants.connect(owner).insertGrant(grantee, "1A", NO_TIMELOCK)
+          ).to.be.revertedWith("Grant already exists");
+        });
       });
 
-      it("A duplicate grant cannot be created", async function () {
-        const { accessGrants, signer1: owner, signer2: grantee } = await loadFixture(deployAndPopulateContractFixture);
+      describe("By anybody through insertGrantBySignature", function () {
+        it("Inserts grant", async function () {
```

Comment:
I'd love to demonstrate it fails with bad sigs.

Reply:
Done.